wcfsetup/install/files/lib/acp/form/LoginForm.class.php

```diff
@@ -213,7 +213,7 @@ public function save() {
 	 */
 	protected function performRedirect(bool $needsMultifactor = false) {
 		if ($needsMultifactor) {
-			$this->url = LinkHandler::getInstance()->getControllerLink(MultifactorAuthenticationForm::class, [
+			$this->url = LinkHandler::getInstance()->getLink('MultifactorAuthentication', [
```

Comment:
I don't like this change, what's the motivation behind this?

Reply:
Using `getLink` will implicitly preserve the current environment. `getControllerLink` forces the environment of the class (in this case: the frontend).